### ISSUE TITLE
[Bug] Image in TableView may display wrong image when scrolling fast

### ISSUE THREAD

**Issue by domenk** [xamarin/Xamarin.Forms#10039]
# Description

When scrolling TableView fast, Image element can display wrong image. Test on Image that retrieves Source from Extension.

Issue occurs only on Android.

This particular issue does not occur on iOS.

_(But iOS has other problems: image won't display if it has no WidthRequest or HeightRequest and if adjacent Label has empty Text. If Label's Text is not empty, Image will sometimes get different dimensions if scrolling fast.)_

### Steps to Reproduce

Run the attached project: [FormsBug.zip](https://github.com/xamarin/Xamarin.Forms/files/4360129/FormsBug.zip)

Scroll TableView by swiping fast. It can take a few tries for issue to appear.

![Wrong image](https://user-images.githubusercontent.com/8338832/77173345-a9e6e280-6abf-11ea-8d2d-a86d25fddd29.gif)

### Expected Behavior

Correct image should be shown.

### Actual Behavior

Wrong image is shown.

### Basic Information

- Version with issue: 4.5.0.356
- Last known good version: /
- IDE: Visual Studio 2019 16.5.0
- Platform Target Frameworks:
  - iOS: 13.14.1.39
  - Android: 10.2.0.100
- Affected Devices: tested on Nexus 4 API 23 and API 28, and iPhone 11 iOS 13.3

**Comment by PureWeen** [comment]
@domenk if you do this same thing with a CollectionView and a DataTemplateSelector do you see the same problem?

**Comment by domenk** [comment]
No, CollectionView works fine. See [FormsBug_CollectionView.zip](https://github.com/xamarin/Xamarin.Forms/files/4393065/FormsBug_CollectionView.zip).
